Content.Server/Zombies/ZombieSystem.Transform.cs

```diff
@@ -232,16 +239,27 @@ public void ZombifyEntity(EntityUid target, MobStateComponent? mobState = null)
             _npc.SleepNPC(target, htn);
 
             //He's gotta have a mind
-            var hasMind = _mind.TryGetMind(target, out var mindId, out _);
+            var hasMind = _mind.TryGetMind(target, out var mindId, out var mind);
             if (hasMind && _mind.TryGetSession(mindId, out var session))
             {
-                //Zombie role for player manifest
+                // Retrieve the user's bans
+                var antagBans = _banManager.GetAntagBans(session.UserId);
+                var antagAllSelection = Loc.GetString("ban-panel-role-selection-antag-all-option");
+
+                // Check if the user has a ban on "Zombie"
+                if (antagBans != null && (antagBans.Contains(antagAllSelection) || antagBans.Contains(zombiecomp.ZombieRoleId)))
+                {
+                    // Ghost the player if they have a "Zombie" ban
+                    _gameTicker.OnGhostAttempt(mindId, true, true, mind);
+                }
+
```

Comment:
Shouldn't this just return here?

Reply:
This one should just keep going after as it sets up the ghostRole and rips there hands off and other zombie comp prep after this.
I did update the param so they cant return to it.
This is a little scuffed as I don't have ghost role bans done, so technically they could just pick up the ghost role for them self right away but this should be ok until that goes in.